app/src/generator/Sidebar/Sidebar.tsx

```diff
@@ -36,29 +34,28 @@ const Zindices = lazy(() => import("generator/Zindices"));
 const Sizes = lazy(() => import("generator/Sizes"));
 
 const Sidebar = () => {
-  const { selectedTheme, selectedMode, colorModes, themes, changeTheme } =
-    useTheme();
+  const { selectedTheme, selectedMode, colorModes, changeTheme } = useTheme();
 
-  const { updateCustomTheme, open } = useContext(GeneratorContext);
+  const { customTheme, updateCustomTheme, open, themeChanges } =
+    useContext(GeneratorContext);
 
-  const [themeName, setThemeName] = useState("customTheme");
+  const [themeName, setThemeName] = useState(customTheme.name);
```

Comment:
If `setThemeName` is only used to update `themeName` when `customTheme.name` changes, then isn't this just the `customTheme.name` itself?

```suggestion
  const themeName = customTheme.name;
```

It works either way, I'm just commenting [because `useEffect`s are evil](https://react.dev/learn/you-might-not-need-an-effect) 😈 (extra re-renders, dependency array, etc.)